### PR TITLE
fix: added handling of notice alarm

### PR DESCRIPTION
**Pull request by arthmoeros**
Given this issue as context:

https://github.com/cloudamqp/terraform-provider-cloudamqp/issues/219

I'm proposing with this PR that notice alarm "auto-imports" into state on resource "creation", and also apply any changes to the existing notice alarm (that the API allows).

### CODE REVIEW

**Comment by dentarg** [comment]
@tbroden84 I think this could work? You see any problems with it?

**Comment by tbroden84** [comment]
Looks good, will try it out a bit later today.

File: cloudamqp/resource_cloudamqp_alarm.go [hunk 1]
```diff
@@ -100,15 +100,36 @@ func resourceAlarmCreate(d *schema.ResourceData, meta interface{}) error {
...
-	if data["id"] != nil {
-		d.SetId(data["id"].(string))
-		log.Printf("[DEBUG] cloudamqp::resource::alarm::create id set: %v", d.Id())
+		if err != nil {
+			return err
+		}
+
+		for _, alarm := range alarms {
+			if alarm["type"] == "notice" {
+				d.SetId(alarm["id"].(string))
```
Comment (tbroden84) [inline cloudamqp/resource_cloudamqp_alarm.go:114]:
```suggestion
				d.SetId(strconv.FormatFloat(alarm["id"].(float64), 'f', 0, 64))
```
When I run this locally the plugin crashes due to wrong interface conversion. Contrary to data["id"] that is converted to a string in our client library, alarm["id"] is a float64 and cannot be asserted with string. For me to get this to work I had to convert it.

File: cloudamqp/resource_cloudamqp_alarm.go [hunk 2]
```diff
@@ -100,15 +100,36 @@ func resourceAlarmCreate(d *schema.ResourceData, meta interface{}) error {
...
+				log.Printf("[DEBUG] cloudamqp::resource::alarm::create retrieved existing alarm id: %v", d.Id())
+				log.Printf("[DEBUG] cloudamqp::resource::alarm::create invoking existing alarm update")
+				return resourceAlarmUpdate(d, meta)
+			}
+		}
+
+		return errors.New(fmt.Sprintf("Couldn't find notice alarm for instance_id %s", d.Get("instance_id")))
```
Comment (tbroden84) [inline cloudamqp/resource_cloudamqp_alarm.go:121]:
```suggestion
		return fmt.Errorf("Couldn't find notice alarm for instance_id %s", d.Get("instance_id"))
```

**Comment by tbroden84** [comment]
Would also suggest adding the notice type check to `resourceAlarmDelete` too. Since our API backend don't allow `notice` alarm to be deleted, it's a mandatory alarm. We can skip sending this request but still remove it from the state file. Something like:
```
if d.Get("type") == "notice" {
  return nil
}
return api.DeleteAlarm(d.Get("instance_id").(int), params)
```

Otherwise like the solution! Hadn't thought about this approach at all.

**Comment by arthmoeros** [comment]
> Would also suggest adding the notice type check to `resourceAlarmDelete` too. Since our API backend don't allow `notice` alarm to be deleted, it's a mandatory alarm. We can skip sending this request but still remove it from the state file. Something like:
> 
> ```
> if d.Get("type") == "notice" {
>   return nil
> }
> return api.DeleteAlarm(d.Get("instance_id").(int), params)
> ```
> 
> Otherwise like the solution! Hadn't thought about this approach at all.

Yeah, I was thinking about it yesterday, will add that change also so destroy only removes it from the state file

**Comment by arthmoeros** [comment]
There @tbroden84 , added handling of delete on notice alarm, thanks for the feedback

**Review by tbroden84 — APPROVED** [review]
Nice! Thanks for the contribution, we will add this in the next release.

**Pull request merged**